templates/components/status.html.twig

```diff
@@ -1,16 +1,45 @@
-{% macro status(text, options) %}
-    {% set _color = options.color ?? 'green' %}
-    {% set _animated = options.animated ?? true %}
-    {% set _lite = options.lite is defined ? options.lite  : false %}
-    {% set _with_dot = options.with_dot is defined ? options.with_dot  : false %}
-    {% set _extraClass = options.extraClass ?? '' %}
-
-    {%- apply spaceless -%}
-        <span class="status status-{{ _color }} {{ _lite ? 'status-lite' : '' }} {{ _extraClass }}">
-        {% if _with_dot %}
-            <span class="status-dot {{ _animated ? 'status-dot-animated' : '' }}"></span>
-        {% endif %}
-            {{ text }}
-        </span>
-    {%- endapply -%}
-{% endmacro %}
+{% macro status(text, options) %}
+    {% import '@Tabler/components/status.html.twig' as macro %}
+    {% import '@Tabler/includes/utils.html.twig' as utils %}
+
+    {% set _color = options.color ?? 'green' %}
+    {% set _animated = (options.animated ?? true) is same as true %}
+    {% set _lite = (options.lite ?? false) is same as true %}
+    {% if options.with_dot is defined %}
+        {% deprecated 'with_dot option is deprecated, use withDot option instead!' %}
+        {% set _withDot = options.with_dot is same as true %}
+    {% else %}
+        {% set _withDot = (options.withDot ?? false) is same as true %}
+    {% endif %}
+
+    {% set _extraClass = options.extraClass ?? '' %}
+    {% set _attr = options.attr ?? {} %}
+
+    {%- apply spaceless -%}
+        <span
+            class="status status-{{ _color }} {% if _lite %}status-lite{% endif %} {{ _extraClass }}"
+            {{ utils.attr_to_html(_attr) }}
+        >
+            {% if _withDot %}
+                {{ macro.status_dot({animated : _animated}) }}
+            {% endif %}
+            {{ text }}
+        </span>
+    {%- endapply -%}
+{% endmacro %}
+
+{% macro status_dot(options) %}
+    {% import '@Tabler/includes/utils.html.twig' as utils %}
+
+    {% set _color = options.color ?? null %}
+    {% set _animated = (options.animated ?? true) is same as true %}
+    {% set _extraClass = options.extraClass ?? '' %}
+    {% set _attr = options.attr ?? {} %}
+
+    {%- apply spaceless -%}

```

Comment:
You know that Twig itself warns about the `spaceless` usage?
https://twig.symfony.com/doc/3.x/filters/spaceless.html

I would remove it.

Reply:
No ?? Why ? I can't find anything about warnings in the doc 🤔 

How are we supposed to do? Put hyphens at every twig functions usages?

Reply:
<img width="1117" alt="Bildschirmfoto 2023-12-11 um 12 16 57" src="https://github.com/kevinpapst/TablerBundle/assets/533162/8c2d0493-a58c-4dd7-9c51-1cc7c5548b03">

I would only use the `{{- -}}` syntax if needed